### PR TITLE
Added gnome sort in Python

### LINKED ISSUE #4816: Gnome Sort in Python
## 💥 Proposal

Category: Sorting Algorithm
Language: Python

Topic: Gnome Sort

It is a simple exchange sorting algorithm similar to insertion sort, except that it moves an element to its proper place by a series of swaps, as in bubble sort. In each iteration the algorithm finds the first place, where two adjacent elements are in the wrong order and swaps them. Afterwards it compares the swapped element with previous elements and swaps them if they are in the wrong order. If adjacent elements are in proper order or the end is reached, it starts a new iteration.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by mehabhalodiya**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Topic: Gnome Sort in Python

Introduction:

It is a simple exchange sorting algorithm similar to insertion sort, except that it moves an element to its proper place by a series of swaps, as in bubble sort. In each iteration the algorithm finds the first place, where two adjacent elements are in the wrong order and swaps them. Afterwards it compares the swapped element with previous elements and swaps them if they are in the wrong order. If adjacent elements are in proper order or the end is reached, it starts a new iteration.

Time/Space Complexity:

Best case time complexity: O(n)
Worst case time complexity: O(n^2)
Space complexity: O(1)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

### CODE REVIEW

File: Python/sort/gnome_sort.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    while i < len(nums):
+        if nums[i - 1] <= nums[i]:
+            i += 1
+        else:
+            nums[i - 1], nums[i] = nums[i], nums[i - 1]
+            i -= 1
+            if (i == 0):
+                i = 1
+
+user_input = input("Enter the numbers separated by a comma: ").strip()
```
Comment (atarax665) [inline Python/sort/gnome_sort.py:21]: Enclose the driver code in ```if __name__ == “__main__”:``` block.

**Review by atarax665 — CHANGES_REQUESTED** [review]
Resolve merge conflict.

**Review by ankitaggarwal23 — COMMENTED** [review]
Code plagiarized from [here](https://www.w3resource.com/python-exercises/data-structures-and-algorithms/python-search-and-sorting-exercise-13.php).

**Comment by mehabhalodiya** [comment]
Hello @ankitaggarwal23 I have not copied it.
Kindly requesting you to open it and I will do required changes, if you want.

**Comment by mehabhalodiya** [comment]
gnomesort

**Comment by mehabhalodiya** [comment]
See @ankitaggarwal23 I have gone through whole algorithm.

File: Python/sort/gnome_sort.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+nums = [int(item) for item in user_input.split(',')]
```
Comment (HarshCasper) [inline Python/sort/gnome_sort.py]: Simplify this statement

File: Python/sort/gnome_sort.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    i = 1
+
+    while i < len(nums):
+        if nums[i - 1] <= nums[i]:
+            i += 1
+        else:
+            nums[i - 1], nums[i] = nums[i], nums[i - 1]
+            i -= 1
+            if (i == 0):
+                i = 1
```
Comment (HarshCasper) [inline Python/sort/gnome_sort.py]: Rewrite this using For-Loop

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Hi @mehabhalodiya 

I understand your point. Since your Code looks too similar, I have asked for some changes. We will verify if this code is plagiarized or whether you understand things fully, by refactoring a few parts

**Comment by mehabhalodiya** [comment]
gnomesort

I have done requested changes as per @HarshCasper.
Kindly review this.

File: Python/sort/gnome_sort.py (hunk 2 above)
Comment (mehabhalodiya) [reply Python/sort/gnome_sort.py]:
```
# in one line: nums = [int(item) for item in user_input.split(',')]
nums = user_input.split(',')
sequence = []
for item in nums:
    sequence.append(int(item))
```

Simplied in the updated PR!

File: Python/sort/gnome_sort.py (hunk 3 above)
Comment (mehabhalodiya) [reply Python/sort/gnome_sort.py]:
```
def gnomeSort(nums):
    for i in range(1, len(nums)):
        while (i != 0 and nums[i] < nums[i - 1]):
            nums[i - 1], nums[i] = nums[i], nums[i - 1]
            i -= 1
```
Done in the updated PR! @HarshCasper

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**